src/components/PolyChart/BiAxialChart/index.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useMemo } from 'react';
+import {
+    Bar,
+    XAxis,
+    YAxis,
+    CartesianGrid,
+    Tooltip,
+    Legend,
+    TooltipFormatter,
+    ResponsiveContainer,
+    Line,
+    ComposedChart,
+} from 'recharts';
+import { IoIosSwap, IoMdClose } from 'react-icons/io';
+import { AiOutlineExpandAlt } from 'react-icons/ai';
+import { compareNumber, isNotDefined, isDefined, _cs, sum } from '@togglecorp/fujs';
+
+import { formatNumber, getPrecision } from '#components/Numeral';
+import Button from '#components/Button';
+import { BiAxialChartSettings, BiAxialData } from '#types';
+import useBasicToggle from '#hooks/useBasicToggle';
+
+import styles from './styles.css';
+
+const categoryTickFormatter = (value: string) => {
+    const words = value.trim().split(/\s+/);
+    if (words.length <= 1) {
+        return value.slice(0, 3).toUpperCase();
+    }
+    return words.map(item => item[0]).join('').toUpperCase();
+};
+
+const valueTickFormatter: TooltipFormatter = (value) => {
+    if (isNotDefined(value)) {
+        return '';
+    }
+    const numberValue = +value;
+    const str = formatNumber(numberValue, true, true, getPrecision(numberValue));
+    return str;
+};
+
+interface BiAxialChartUnitProps<T> {
+    settings: BiAxialChartSettings<T>;
+    data: T[] | undefined;
+    className?: string;
+    chartClassName?: string;
+    headerClassName?: string;
+    hideActions?: boolean;
+    onDelete: (name: string | undefined) => void;
+    onExpand: (name: string | undefined) => void;
+    expandableIconHidden: boolean;
+}
+
+const chartMargin = {
+    top: 10,
+    right: 10,
+    bottom: 10,
+    left: 10,
+};
+
+export function BiAxialChartUnit<T extends object>(props: BiAxialChartUnitProps<T>) {
+    const {
+        className,
+        chartClassName,
+        headerClassName,
+        settings,
+        data,
+        onDelete,
+        hideActions,
+        onExpand,
+        expandableIconHidden,
+    } = props;
+
+    const [chartTypeToggled, , , onToggleChartType] = useBasicToggle();
+
+    const {
+        title,
+        keySelector,
+        chartData,
+        limit,
+        id,
+    } = settings;
+
+    const formattedChartData: BiAxialData<T>[] = useMemo(() => {
+        if (chartTypeToggled) {
+            const [firstData, secondData] = chartData;
+            const tmpFirstDataType = firstData.type === 'bar' ? 'line' : 'bar';
+            const tmpSecondDataType = secondData.type === 'line' ? 'bar' : 'line';
+            return [
+                { ...firstData, type: tmpFirstDataType },
+                { ...secondData, type: tmpSecondDataType },
+            ];
```

Comment:
What if we try to other line charts later on?
This approach looks okay for now but an easier method would be to just toggle them using map

```suggestion
            const [firstData, secondData] = chartData;
            const tmpFirstDataType = firstData.type === 'bar' ? 'line' : 'bar';
            const tmpSecondDataType = secondData.type === 'line' ? 'bar' : 'line';
            return chartData.map((item) => ({ ...item, type: item.type === 'bar' ? 'line' : 'bar' }));
```